pkg/ipcache/ipcache.go

```diff
@@ -189,6 +192,25 @@ const (
 	Delete CacheModification = "Delete"
 )
 
+func keyToIP(key string) (net.IP, error) {
+	requiredPrefix := fmt.Sprintf("%s/", path.Join(IPIdentitiesPath, AddressSpace))
+	if !strings.HasPrefix(key, requiredPrefix) {
+		return nil, fmt.Errorf("Found invalid key %s outside of prefix %s", key, IPIdentitiesPath)
+	}
+
+	suffix := strings.TrimPrefix(key, requiredPrefix)
+	if suffix[0] == '/' {
```

Comment:
Can this ever happen if you trim the prefix with `/`?

Reply:
Good point, will remove.